src/position/major.js

```diff
@@ -0,0 +1,120 @@
+import * as Digit from './digit.js'
+import * as Digits from './digits.js'
+import * as Minor from './minor.js'
+import { base52 as base } from './digits.js'
+import * as Base from './base.js'
+
+import { EQUAL, CONSECUTIVE } from './digits.js'
+
+const [[outerNegative, innerNegative], [innerPositive, outerPositive]] =
+  base.ranges
+
+export { base }
+export const MIN = Base.min(base)
+export const MAX = Base.max(base)
+
+export const min = new Uint8Array([MIN, ...new Uint8Array(MIN).fill(Minor.MIN)])
+export const max = new Uint8Array([MAX, ...new Uint8Array(MAX).fill(Minor.MAX)])
+export const zero = () => innerPositive
+
+export { EQUAL, CONSECUTIVE }
+
+/**
+ * Major byte describes the size range of the minor positions. It highly
+ * recommended to use only bytes in the  'A-Za-z' character code range meaning
+ *
+ * ```
+ * [65...90,  97..122]
+ * ```
+ *
+ * Major codes like `z:65` and `a:97` denote two digit `Minor` range, codes `Y:66`
```

Comment:
These codes *do not* line up with their ASCII representations, IIUC -- `A:65`, not `z`, no? `b:98` and `B:66` as well.

Reply:
IIUC what's the difference between `A:65` and `a:97` if both signify minor patch size, this seems to have a lot of opinions on underlying encoding if "A" and "a" have some significant relationship other than its relative position

Reply:
> These codes _do not_ line up with their ASCII representations, IIUC -- `A:65`, not `z`, no? `b:98` and `B:66` as well.

You're right thanks for caching these. Fix in the suggestion



> IIUC what's the difference between `A:65` and `a:97` if both signify minor patch size, this seems to have a lot of opinions on underlying encoding if "A" and "a" have some significant relationship other than its relative position

Yeah we treat [A..Z] as a top segment (for insertions in the top of the list) and [a..z] range as bottom segment (for insertions in the bottom of the list). Both ranges use shorter sizes and increasingly long ones towards the edges. This way key size increase logarithmically